subsys/net/lib/lwm2m/lwm2m_rw_senml_json.c

```diff
@@ -0,0 +1,1609 @@
+				ret = -EINVAL;
+				goto end_of_operation;
+			}
+
+			base_name[fd.value_len] = '\0';
+			/* Relative name is optional - preinitialize full name with base name */
+			snprintk(full_name, sizeof(full_name), "%s", base_name);
+			ret = parse_path(full_name, strlen(full_name), &resource_path);
+			if (ret < 0) {
+				LOG_ERR("Relative name too long");
+				ret = -EINVAL;
+				goto end_of_operation;
+			}
+			resource_path.level = ret;
+
+			if (resource_path.level) {
+				path_valid = true;
+			}
+
+			if (block_ctx) {
+				block_ctx->base_name_path = resource_path;
+				block_ctx->base_name_stored = true;
+			}
+
+			break;
+		case SENML_JSON_NAME_ATTRIBUTE:
+
+			/* handle resource name */
+			if (fd.value_len > MAX_RESOURCE_LEN) {
+				LOG_ERR("Relative name too long");
+				ret = -EINVAL;
+				goto end_of_operation;
+			}
+
+			/* get value for relative path */
+			if (buf_read(name, fd.value_len, CPKT_BUF_READ(msg->in.in_cpkt),
+				     &fd.value_offset) < 0) {
+				LOG_ERR("Error parsing relative path!");
+				ret = -EINVAL;
+				goto end_of_operation;
+			}
+
+			name[fd.value_len] = '\0';
+
+			/* combine base_name + name */
+			snprintk(full_name, sizeof(full_name), "%s%s", base_name, name);
+			ret = parse_path(full_name, strlen(full_name), &resource_path);
+			if (ret < 0) {
+				LOG_ERR("Relative name too long");
+				ret = -EINVAL;
+				goto end_of_operation;
+			}
+			resource_path.level = ret;
+
+			if (block_ctx) {
+				//Store Resource data Path to base name path but store separately path level
+				uint8_t path_level = block_ctx->base_name_path.level;
+				block_ctx->base_name_path = resource_path;
+				block_ctx->resource_path_level = resource_path.level;
+				block_ctx->base_name_path.level = path_level;
+				block_ctx->full_name_true = true;
+			}
+			path_valid = true;
+			break;
+
+		case SENML_JSON_FLOAT_VALUE_ATTRIBUTE:
+		case SENML_JSON_BOOLEAN_VALUE_ATTRIBUTE:
+		case SENML_JSON_OBJ_LINK_VALUE_ATTRIBUTE:
+		case SENML_JSON_OPAQUE_VALUE_ATTRIBUTE:
+		case SENML_JSON_STRING_VALUE_ATTRIBUTE:
+		case SENML_JSON_STRING_BLOCK_DATA:
+			//Opaque stream end detetcted
+			data_value = true;
+			break;
+
+		case SENML_JSON_UNKNOWN_ATTRIBUTE:
+			LOG_ERR("Unknow attribute");
+			ret = -EINVAL;
+			goto end_of_operation;
+			break;
+		default:
+			break;
+		}
+
+		if (data_value && path_valid) {
+			/* parse full_name into path */
+			if (block_ctx) {
+				//Store json Flgs
+				block_ctx->json_flags = fd.json_flags;
+			}
+
+			msg->path = resource_path;
+			ret = lwm2m_senml_write_operation(msg, &fd);
+
+			if (ret < 0) {
+				break;
+			}
+		}
+	}
+	//Do we have a data value which is part of the Coap blocking process
+	if ((fd.json_flags & T_VALUE) && !(fd.json_flags & T_OBJECT_END) && !data_value &&
+	    block_ctx && fd.value_len) {
+		if (!path_valid) {
+			LOG_ERR("No path available for Coap Block sub sequency");
+			ret = -EINVAL;
+			goto end_of_operation;
+		}
+		//Store Json File description flags
```

Comment:
In this file, you have quite many `//` comments mixed with `/*  */` comment blocks.
Stick to one style, which I think should be `/* */`

Reply:
Ok I fix those.